store/store_asserts.go

```diff
@@ -107,6 +99,68 @@ func (s *Store) Assertion(assertType *asserts.AssertionType, primaryKey []string
 	return asrt, nil
 }
 
+// SeqFormingAssertion retrieves the sequence-forming assertion for the given
+// type (currently validation-set only).
+func (s *Store) SeqFormingAssertion(assertType *asserts.AssertionType, sequenceKey []string, sequence int, user *auth.UserState) (asserts.Assertion, error) {
+	if !assertType.SequenceForming() {
+		return nil, fmt.Errorf("internal error: requested non sequence-forming assertion type %q", assertType.Name)
+	}
+	v := url.Values{}
+	v.Set("max-format", strconv.Itoa(assertType.MaxSupportedFormat()))
+
+	hasSequenceNumber := sequence > 0
+	if hasSequenceNumber {
+		// full primary key passed, query specific sequence number.
+		v.Set("sequence", fmt.Sprintf("%d", sequence))
+	} else {
+		// query for the latest sequence.
+		v.Set("sequence", "latest")
+	}
+	u := s.assertionsEndpointURL(path.Join(assertType.Name, path.Join(sequenceKey...)), v)
+
+	var asrt asserts.Assertion
+
+	err := s.downloadAssertions(u, func(r io.Reader) error {
+		// decode assertion
+		dec := asserts.NewDecoder(r)
+		var e error
+		asrt, e = dec.Decode()
+		return e
+	}, func(svcErr *assertionSvcError) error {
+		// error-list indicates v2 error response.
+		if svcErr.isNotFound() {
+			// XXX: this re-implements asserts.HeadersFromPrimaryKey() but is
+			// more relaxed about key length, making sequence optional. Should
+			// we make it a helper on its own in store for the not-found-error
+			// handling?
```

Comment:
yes, I think a new helper HeadersFromSequenceKey would be alright

Reply:
I'll address this in a followup.